### ISSUE TITLE
augmentation: Resize, LongestMaxSize, SmallestMaxSize and slice-mode RandomResizedCrop record a corner-aligned matrix for an align_corners=False image

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4804]
## Summary

With `align_corners=False`, `Resize`, `LongestMaxSize`, `SmallestMaxSize` and slice-mode `RandomResizedCrop` record a `transform_matrix` that does not match the image they produce. The matrix maps corner to corner, `x' = x * (W_out - 1) / (W_in - 1)`, which is correct for `align_corners=True`. The image comes from `F.interpolate`, and at `align_corners=False` that samples on the half-pixel grid, `x' = (x + 0.5) * W_out / W_in - 0.5`.

Inside `AugmentationSequential`, keypoints and boxes follow the matrix. With `align_corners=False` they therefore end up away from the content they annotate, by up to half an output pixel toward the borders. The same flag also produces different images in the two `RandomResizedCrop` cropping modes: resample mode warps with the matrix, and `align_corners` there changes only the out-of-range normalization.

Every other 2D geometric class I checked keeps the image and the matrix in agreement at both settings (error below 3e-15 px in float64). The checked classes are Rotation, Rotation90, Affine, Perspective, Shear, Translate, the flips, PadTo, CenterCrop resample mode and RandomResizedCrop resample mode. All of these four classes default to `align_corners=True`, so a call is affected only when it passes `align_corners=False` explicitly. The 3D analogue of a flag-dependent normalization mismatch is #4503. The split between the defaults is #4412.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print("kornia", kornia.__version__, "torch", torch.__version__)
# A float64 ramp whose value is its own x: an output pixel's value is the source x it sampled.
ramp = torch.arange(7.0, dtype=torch.float64).expand(1, 1, 5, 7).contiguous()
for cls, args in ((K.Resize, ((10, 14),)), (K.LongestMaxSize, (14,)), (K.SmallestMaxSize, (10,))):
    for ac in (True, False):
        aug = cls(*args, align_corners=ac, p=1.0)
        out = aug(ramp)
        inv = torch.linalg.inv(aug.transform_matrix[0])
        u = torch.arange(out.shape[-1], dtype=torch.float64)
        from_matrix = inv[0, 0] * u + inv[0, 2]  # source x that the recorded matrix assigns to output column u
        inner = (from_matrix > 0.5) & (from_matrix < 5.5)  # away from the clamped border columns
        err = (out[0, 0, 2][inner] - from_matrix[inner]).abs().max().item()
        print(f"{cls.__name__:15s} align_corners={ac!s:5s} max |sampled x - matrix x| = {err:.4f} source px")
torch.manual_seed(0)
x = torch.rand(1, 1, 20, 24)
for ac in (True, False):
    torch.manual_seed(1)
    s = K.RandomResizedCrop((9, 13), align_corners=ac, p=1.0, cropping_mode="slice")
    a = s(x)
    r = K.RandomResizedCrop((9, 13), align_corners=ac, p=1.0, cropping_mode="resample")(x, params=s._params)
    print(f"RandomResizedCrop align_corners={ac!s:5s} max |slice - resample|, same params = {(a - r).abs().max().item():.4f}")
```

Output (CPU):

```
kornia 0.9.0rc1 torch 2.14.0
Resize          align_corners=True  max |sampled x - matrix x| = 0.0000 source px
Resize          align_corners=False max |sampled x - matrix x| = 0.1731 source px
LongestMaxSize  align_corners=True  max |sampled x - matrix x| = 0.0000 source px
LongestMaxSize  align_corners=False max |sampled x - matrix x| = 0.1731 source px
SmallestMaxSize align_corners=True  max |sampled x - matrix x| = 0.0000 source px
SmallestMaxSize align_corners=False max |sampled x - matrix x| = 0.1731 source px
RandomResizedCrop align_corners=True  max |slice - resample|, same params = 0.0000
RandomResizedCrop align_corners=False max |slice - resample|, same params = 0.4456
```

The ramp's value at an output pixel is the source `x` that pixel sampled. The matrix predicts the same `x` exactly at `align_corners=True`. At `False` it is off by up to 0.17 source px here, which is 0.35 output px at this 2x upscale.

## Possible fixes

- Record the half-pixel matrix, `x' = (x + 0.5) * s - 0.5`, when these paths pass `align_corners=False` to `interpolate`.
- Or make the slice/resize paths sample corner-aligned whatever the flag. That would match what the flag means for the warp-based classes, where it changes only the out-of-range normalization.

Found while reviewing #4791, where the geometric base states that `transform_matrix` maps input pixel coordinates to output pixel coordinates.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
